src/_common/content/content-editor/embeds/youtube-video.ts

```diff
@@ -30,23 +42,16 @@ export class YouTubeVideoEmbed extends EmbedSource {
 			return false;
 		}
 
-		const url = this.isValidLink(link);
-		if (url !== false && /(.+\.)?youtube\.com/i.test(url.hostname)) {
-			const videoId = url.searchParams.get('v');
-			if (videoId !== null && videoId.length === 11) {
-				return videoId;
-			}
-		}
-
-		return false;
+		return this.isValidLink(link);
 	}
 
 	isValidLink(link: string) {
-		try {
-			return new URL(link);
-		} catch (error) {
-			// Swallow error. new Url throws on invalid URLs, which can very well happen.
-			return false;
+		const results = getYoutubeVideoId(link) || false;
+
+		if (results && results.length === 11) {
```

Comment:
we shouldn't check the length, it might change in future

Reply:
Also, isValidLink should probably just return a boolean.
You can call getYoutubeVideoId above in the getLinkSource directly